File: app/backend/routes/system.py
# app/backend/routes/system.py
from flask import Blueprint, current_app
from ..responses import success

system_bp = Blueprint("system", __name__)


@system_bp.route("/api/system/status")
def get_system_status():
    config = current_app.config.get("BACKEND_CONFIG", {})
    raw = current_app.config.get("LAN_ADDRESSES", [])
    lan_addresses = [a for a in raw if not a.startswith("127.0.0.1:")]
    return success(
        data={
            "status": "running",
            "version": config.get("version", "unknown"),
            "started_at": current_app.config.get("STARTED_AT", ""),
            "lan_addresses": lan_addresses,
        }
    )

File: app/frontend/src/api/system.ts
import { apiRequest } from './client';

export interface SystemStatus {
  status: 'running' | 'error' | string;
  version: string;
  started_at: string;
  lan_addresses: string[];
  message?: string;
}

export function getSystemStatus() {
  return apiRequest<SystemStatus>('/api/system/status');
}
